packages/eslint-plugin/docs/rules/no-floating-promises.mdx

```diff
@@ -13,13 +13,13 @@ A "floating" Promise is one that is created without any code set up to handle an
 Floating Promises can cause several issues, such as improperly sequenced operations, ignored Promise rejections, and more.
 
 This rule reports when a Promise is created and not properly handled.
-Valid ways of handling a Promise-valued statement include:
+Valid ways of silencing a Promise-valued statement include:
 
-- `await`ing it
-- `return`ing it
-- `void`ing it
-- Calling its `.then()` with two arguments
-- Calling its `.catch()` with one argument
+- `await`ing it (❌ unhandled)
+- `return`ing it (❌ unhandled)
+- [`void`ing it](#ignorevoid) (❌ unhandled)
+- Calling its `.then()` with two arguments (✅ handled)
+- Calling its `.catch()` with one argument (✅ handled)
```

Comment:
I understand why you'd want to have these `(❌ unhandled)`/`(✅ handled)` indicators, but I'm not sure they're an improvement here, without a deeper dive into unhandled rejections. 

The reason I say this is that code like `promise.catch(() => { throw new Error('unhandled!') });` _is_ still unhandled, but (intentionally) not flagged by the rule.

Furthermore, `await`ing or `return`ing promises _do_ prevent a unhandled rejections (assuming the caller is handled appropriately).

For now, my thought would be to leave these markers off (but maybe this can be revisited when we do futher docs improvements for the promise rules, which is already on our redar).

Reply:
I see your point. I'd ask you why this is intentional, but I guess that would be pointless.

`await`ing and `return`ing do not prevent unhandled rejections but delegate the handling to, respectively a `try-catch` and the parent function.

As you surely have seen, these markers are meant also for the `ignoreVoid` section warning. So I wonder if there's another way to include such markers in a softer way, maybe by preparing it for the change you mentioned...

Reply:
I've removed the markers as I've improved the message in `ignoreVoid`.
Let me know if it is okay to you

Reply:
> I'd ask you why this is intentional, but I guess that would be pointless

Hardly! 🙂 The short version is just that it's impossible to know that a function won't throw in JS, so, in reality, `.then()` and `.catch()` can never guaranteeably prevent a floating promise. 

For example, `promise.catch(() => maybeThrows())` can never be known not to throw, so _technically_ we could mark it as unhandled. But then, _everything_ would be unhandled! So instead, we say, `promise.catch(() => maybeThrows())` _looks_ like it's handled, so the rule should not report.

Similarly, for `promise.then(() => onFulfilled(), () => onRejected())`, we know that `onFulfilled` _could_ throw, and isn't any different than `promise.then(() => onFulFilled())`, which we _would_ mark as unhandled. However, because the `onRejected` handler is provided, we say that the usage looks like the developer was aware of rejections when they wrote the code, and, therefore, it it _looks_ handled. 

So, no-floating-promises really becomes about putting up reasonable guardrails, rather than attempting to make guarantees.

There are quite a few issues that discuss these tradeoffs.

Reply:
The fact it "looks like it handled" could be a thing worth to be specified in a new paragraph about handling promises.